### PR TITLE
Reshuffle of the styling helpers

### PR DESCRIPTION
**Pull request by dzearing**
We recently added the exports for the styling helpers to the Styling package. Unfortunately this added React as a dependency of Styling, which is a dependency of Icons, which means you end up pulling in React just to get an icon classname, which is in some PLT scenarios.

As a short term fix, we need to move them to utilities, which already has the React dependency.

A better approach would be to separate react utilities from js utilities, but in the end, as long as consumers are consuming fabric-react components, they're going to get react anyway, so it's not a huge concern.

This is an unfortunate breaking change, but given the helpers were only recently introduced, we want these changes in master so that we can get them into products using 5.0.

This means instead of these:

```tsx
import {
  styled,
  IStyleFunction,
  classNameFunction
} from 'office-ui-fabric-react/lib/Styling';
```

you would get them from utilities:

```tsx
import {
  styled,
  IStyleFunction,
  classNameFunction
} from 'office-ui-fabric-react/lib/Utilities';
```

No api changes, just export changes.

### CODE REVIEW

File: apps/fabric-website/gulpfile.js [hunk 1]
```diff
@@ -1,73 +0,0 @@
-'use strict';
-
-let build = require('@microsoft/web-library-build');
-let gulp = require('gulp');
```
Comment (christiango) [inline apps/fabric-website/gulpfile.js]: seems like there are some unrelated changes in this PR?

Comment (dzearing) [reply apps/fabric-website/gulpfile.js]: just deleting the old gulp files while im here. Also included some missing typings.

Comment (christiango) [reply apps/fabric-website/gulpfile.js]: ok, just wanted to make sure it was unrelated to the PR
